### DISCUSSION TITLE
编辑器生成表单之后，最终的形态是什么？

### DISCUSSION THREAD

**Discussion by dahui4dev** [alibaba/designable#4] [Q&A] [answered]
有了编辑器，可以可视化的搭建表单页面，当吧表单页面搭建好之后，这个搭建好的表单将以何种形态在前端展示？

分以下几种情况：

1. 保存：将表单的 schema 直接保存，使用的时候直接将 schema 应用于 `<Field schema={schema}>`;
2. 发布：将搭建好的表单 schema 提交后端，后端可通过提交的 schema 构建好表单页面（这个可以是纯 html、或者 js），返回给前端展示；

不知道规划中的使用方式是哪种？或者是其他的形态。

**Comment by janryWang** [comment, marked as answer]
> 有了编辑器，可以可视化的搭建表单页面，当吧表单页面搭建好之后，这个搭建好的表单将以何种形态在前端展示？
> 
> 分以下几种情况：
> 
> 1. 保存：将表单的 schema 直接保存，使用的时候直接将 schema 应用于 `<Field schema={schema}>`;
> 2. 发布：将搭建好的表单 schema 提交后端，后端可通过提交的 schema 构建好表单页面（这个可以是纯 html、或者 js），返回给前端展示；
> 
> 不知道规划中的使用方式是哪种？或者是其他的形态。

其实就是个JSON Schema生产器而已，没啥特别的形态

**Reply by dahui4dev** [reply]
也就是说，编辑器只是生产 JSON Schema ，想用搭建好的表单，其实还是需要写代码。要把 `<Field schema={schema}>` 写好，把生成的 Schema 塞到代码里。而不是将生成好的表单直接给用户用？

**Reply by janryWang** [reply]
不会，你说的场景是更上层的封装

**Reply by dahui4dev** [reply]
清楚了，你这个分层分的相当细。👍

**Reply by dingyanqiang** [reply]
JSON Schema生产器，那和Xrender的playground有什么差异呢？

**Reply by jinphen** [reply]
> JSON Schema生产器，那和Xrender的playground有什么差异呢？

xrender的playground就是基于这个designable去做的，我的理解是designable是一个更通用的配置生成器。

**Reply by simdd** [reply]
> 清楚了，你这个分层分的相当细。👍

那预览能力，dn 提供吗？

**Comment by zhenghuahou** [comment]
我也有这个疑问，
<img width="744" alt="image" src="https://github.com/alibaba/designable/assets/3623485/9ac33b8a-46af-4abc-a9fb-6e8b02aecd05">
拖拽生成的这个页面，框架上是否支持配置，让单独生成一个预览视图呢？
